gomock/matchers.go

```diff
@@ -122,3 +123,86 @@ func Not(x interface{}) Matcher {
 func AssignableToTypeOf(x interface{}) Matcher {
 	return assignableToTypeOfMatcher{reflect.TypeOf(x)}
 }
+
+type compositeMatcher struct {
```

Comment:
nit: Can we rename this to `allMatcher` to better match the existing convention?